ResearchKit/Consent/ORKVisualConsentStepViewController.m

```diff
@@ -251,10 +252,20 @@ - (IBAction)goToPreviousPage {
 }
 
 - (IBAction)next {
-    [self showViewController:[self viewControllerForIndex:[self currentIndex]+1] forward:YES animated:YES];
-    ORKAccessibilityPostNotificationAfterDelay(UIAccessibilityScreenChangedNotification, nil, 0.5);
+    ORKConsentSceneViewController *currentConsentSceneViewController = [self viewControllerForIndex:[self currentIndex]];
+    [currentConsentSceneViewController setScrollEnabled:NO];
+    [currentConsentSceneViewController scrollToTopAnimated:YES completion:^(BOOL finished) {
+        [self showNextViewController];
```

Comment:
My one hesitation is here - couldn't something interrupt this animation? If the animation is not finished, then it seems like the transition has been interrupted, and we should not transition, and we re-enable animation the current view controller. I'm thinking of someone tapping the Cancel button, which could interrupt the animation, for example.